.github/workflows/code-tidy.yml

```diff
@@ -0,0 +1,73 @@
+name: Tidy Checks
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  RunClangTidy:
+    runs-on: ubuntu-24.04
+    steps:
+
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        submodules: 'recursive'
+
+    - name: Install dependencies
+      run: |
+        sudo ./scripts/uni-get-dependencies.sh
+        sudo apt -qq -y install clang-tidy
+
+    - name: Run CMake
+      run: cmake -S . -B build -GNinja
+
+    - name: Build autogenerated files
+      run: ninja -C build CMakeFiles/OpenSCAD_autogen
+
+    - name: Create Cache Timestamp
+      id: cache_timestamp
+      uses: nanzm/get-time-action@v2.0
+      with:
+        format: 'YYYY-MM-DD-HH-mm-ss'
+
+    - name: Retrieve cached results
+      uses: actions/cache@v4
+      with:
+        path: /home/runner/.cache/clang-tidy
```

Comment:
The name of the directory immediately below this one contains a hash of the clang-tidy version used. Does that mean we may need to clean out this from time to time as well? If so, perhaps add a TODO comment to that tune.

Reply:
Yes, things will accumulate, albeit probably pretty slow. Added a comment.